### PR TITLE
CRM-16145, CRM-16246 - Aggregate AngularJS modules. De-aggregate ckeditor (fixing ckeditor with both Drupal aggregation and this aggregation).

### PR DESCRIPTION
**Pull request by totten**

### CODE REVIEW

**Comment by totten** [comment]
This works for me in Firefox and Chrome, whether Drupal's aggregation is enabled or disabled.

File: Civi/Angular/Page/Main.php [hunk 1]
```diff
@@ -82,17 +82,33 @@ public function registerResources() {
...
+        foreach ($this->angular->getResources($moduleName, 'js', 'cacheUrl') as $url) {
+          $this->res->addScriptUrl($url, self::DEFAULT_MODULE_WEIGHT + (++$headOffset), 'html-header');
+          // addScriptUrl() bypasses the normal string-localization of addScriptFile(),
+          // but that's OK because all Angular strings (JS+HTML) will load via crmResource.
+        }
       }
     }
+    else {
+      // Note: addScriptUrl() bypasses the normal string-localization of addScriptFile(),
+      // but that's OK because all Angular strings (JS+HTML) will load via crmResource.
```
Comment (colemanw) [inline Civi/Angular/Page/Main.php]: If the only reason for using `addScriptUrl` is to bypass string parsing, `addScriptFile` does have a param for that.

Comment (totten) [inline Civi/Angular/Page/Main.php]: _For the aggregated/non-debug case_ -  addScriptFile('civicrm','foo.js') would produce a URL like "http://example.com/sites/all/modules/civicrm/foo.js" (pointing to a physical file based on the module's resource URL), but we want a URL like "http://example.com/civicrm/ajax/angular-modules?format=js" (pointing to virtual file).

 _For the separated/debug case_ - That might have been the original reason. At the moment, it's worth noting that getResources() is already returning URLs.

Comment (colemanw) [inline Civi/Angular/Page/Main.php]: Makes sense.

**Pull request merged**